src/dra/42398.c

```diff
@@ -47,16 +56,193 @@ void func_800E2398(const char* str) {
+    const int MaxScreenshotCount = 1000;
+    const int BytesPerPixel = 3;
+    char fileName[0x100];
+    u8 bmp[0x100];
+    s8 buffer[0x30];
+    s32 fid;
+    s32 width;
+    s32 height;
+    u32 fileSize;
+    s32 bufferPos;
+    s8* dst;
+    u16* src;
+    s32 i;
+    s32 j;
+
+    if (!(g_pads[1].tapped & PAD_SELECT && g_pads[1].pressed & PAD_UP)) {
+        return;
+    }
+
+    StoreImage(&g_CurrentBuffer->disp.disp, (u32*)0x80200000);
+    DrawSync(0);
+
+    for (i = 0; i < MaxScreenshotCount; i++) {
+        __builtin_memcpy(fileName, D_800DB3B8, sizeof(D_800DB3B8));
+        fileName[14] += i / 100;
+        fileName[15] += i / 10 - i / 100 * 10;
+        fileName[16] += i % 10;
+        fid = open(fileName, O_RDONLY);
+        if (fid < 0) {
+            break;
+        }
+        close(fid);
+    }
+
+    fid = open(fileName, O_CREAT);
+    if (fid < 0) {
+        FntPrint(D_800DB3D0, &fileName);
+        return;
+    }
 
-INCLUDE_ASM("asm/us/dra/nonmatchings/42398", func_800E2824);
+    width = g_CurrentBuffer->disp.disp.w;
+    height = g_CurrentBuffer->disp.disp.h;
+    for (i = 0; i < BmpHeaderLen; i++) {
+        bmp[i] = 0;
+    }
+    fileSize = BmpHeaderLen + width * height * BytesPerPixel;
+    bmp[0x00] = 'B';               // Bitmap signature
+    bmp[0x01] = 'M';               // for Windows OS
+    bmp[0x02] = fileSize;          // total file size of
+    bmp[0x03] = fileSize >> 8;     // the bitmap file
+    bmp[0x04] = fileSize >> 16;    // in little endian
+    bmp[0x05] = fileSize >> 24;    //
+    bmp[0x0A] = BmpHeaderLen;      // Data offset
+    bmp[0x0E] = 40;                // header size
+    bmp[0x1A] = 1;                 // number of color planes
+    bmp[0x1C] = BytesPerPixel * 8; // bits per pixel
+    bmp[0x12] = width;
+    bmp[0x13] = width / 256;
+    bmp[0x16] = height;
+    bmp[0x17] = height / 256;
+    if (write(fid, &bmp, BmpHeaderLen) < 0) {
+        FntPrint(D_800DB3DC);
+        return;
+    }
+
+    bufferPos = 0;
+    dst = buffer;
+    src = 0x80200000;
+    for (i = height - 1; i >= 0; i--) {
+        u16* start = buffer;
+        for (j = 0; j < width; j++) {
+            u16 pixelColor = src[j + i * width];
+            *dst++ = (pixelColor >> 7) & 0xF8; // B
+            *dst++ = (pixelColor >> 2) & 0xF8; // G
+            *dst++ = (pixelColor & 0x1F) << 3; // R
+            if (++bufferPos == 0x10) {
+                if (write(fid, start, bufferPos * BytesPerPixel) < 0) {
+                    FntPrint(D_800DB3DC);
+                    return;
+                }
+                bufferPos = 0;
+                dst = start;
+            }
+        }
+    }
+
+    if (bufferPos != 0) {
+        if (write(fid, &buffer, bufferPos * BytesPerPixel) < 0) {
+            FntPrint(D_800DB3DC);
+            return;
+        }
+    }
+    if (close(fid) < 0) {
+        FntPrint(D_800DB3E4);
+        return;
+    }
+    DebugInputWait(fileName);
+}
+
+void DebugCaptureVideo(void) {
```

Comment:
I think this should be DebugCaptureDemo, a video is something different. I'm assuming this was used to capture the strategy demos for defeating the bosses? That's what japan devs usually call Demos.